### ISSUE TITLE
salt zfs scheduled_snapshot uses gmtime() instead of localtime

### ISSUE THREAD

**Issue by angeloudy** [saltstack/salt#43971]
### Setup
OS: FreeBSD 11.1
Salt version: py27-salt-2017.7.1_1

### Steps to Reproduce Issue
Use the following sls to create snapshots
  zfs.scheduled_snapshot:
    - name: storage/test
    - prefix: test
    - schedule:
        hour: 168
        day: 60 
        month: 9999 
        year: 999
One hour later, run it again, no new snapshots created.

By checking the code, I found
```

   790         # create snapshot
    791         needed_holds = []
    792         current_timestamp = gmtime()
    793         for hold in snapshots:
    794             # check if we need need to consider hold
    795             if schedule[hold] == 0:
    796                 continue
    797
```

I think this is caused by the following line:
`current_timestamp = gmtime()`

After I changed it to 
`current_timestamp = localtime()`
It worked.

### Versions Report
Salt Version:
           Salt: 2017.7.1

Dependency Versions:
           cffi: 1.7.0
       cherrypy: unknown
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: 2.0.2
      gitpython: 2.1.5
          ioflo: Not Installed
         Jinja2: 2.9.5
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: 1.2.5
      pycparser: 2.10
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.13 (default, Aug 11 2017, 18:52:24)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 2.0.3
        timelib: Not Installed
        Tornado: 4.5.2
            ZMQ: 4.2.2

System Versions:
           dist:
         locale: US-ASCII
        machine: amd64
        release: 11.1-RELEASE-p1
         system: FreeBSD
        version: Not Installed

**Comment by garethgreenaway** [comment]
@angeloudy Thanks for the report.  We would of course welcome a PR with the fix for this issue.

**Comment by angeloudy** [comment]
I have created a pull request.

**Comment by rallytime** [comment]
Closed via #43996 and #44502
